src/BenchmarksClient/Startup.cs

```diff
@@ -96,8 +96,13 @@ public static int Main(string[] args)
                 var job = allJobs.FirstOrDefault();
                 if (job != null)
                 {
-                    var jobLogText = $"[ID:{job.Id} Connections:{job.Connections} Threads:{job.Threads} " +
-                        $"Duration:{job.Duration} Pipeline:{job.PipelineDepth} Method:{job.Method}";
+                    var jobLogText =
+                        $"[ID:{job.Id} Connections:{job.Connections} Threads:{job.Threads} Duration:{job.Duration} Method:{job.Method}";
+
+                    if (!string.IsNullOrEmpty(job.ScriptName) && job.PipelineDepth > 0)
```

Comment:
_Current_ scripts accept (but do not require) a pipeline depth argument. Even with that optionality, the restriction you're enforcing here doesn't make sense in general. What about future scripts with new arguments?

My proposed assertion would check that a non-default pipeline depth has a corresponding script to do something with it. Not a requirement from my perspective but somewhat in line with what you've written.

> I could make it so if pipeline depth <= 0 then default to 1.

Both existing scripts already do that.